packages/x-date-pickers-pro/src/SingleInputDateRangeField/tests/editing.SingleInputDateRangeField.test.tsx

```diff
@@ -63,10 +63,10 @@ describe('<SingleInputDateRangeField /> - Editing', () => {
 
           // Start date
           userEvent.keyPress(input, { key: keyToClearValue });
-          expect(handleChange.callCount).to.equal(0);
+          expect(handleChange.callCount).to.equal(1);
           userEvent.keyPress(input, { key: 'ArrowRight' });
           userEvent.keyPress(input, { key: keyToClearValue });
-          expect(handleChange.callCount).to.equal(1);
+          expect(handleChange.callCount).to.equal(2);
```

Comment:
OK so if I understand correctly, your 1st proposal would be to add a check that says "do not fire if the previous firing was with a partially filled date and the new firing is also with a partially filled date".

I would be interested to have the opinion of @sebastian-ludwig
And if we can't reach to an agreement on when `onChange` should be called, see how we can make that customizable.

Reply:
> OK so if I understand correctly, your 1st proposal would be to add a check that says "do not fire if the previous firing was with a partially filled date and the new firing is also with a partially filled date".

Yes, both of my solutions are in essence the addition of comparison between the previous date and the currently changed one, if they do not differ (`null === null` and `'Invalid date' === 'Invalid date'`), why bother spamming `onChange` events and create the same problem we currently have when adding a somewhat simple check should be sufficient in order to avoid redundant emitting. 🤔 
The difference between the solutions is whether `'Invalid date' === null` or not. 🤔 
Because going with the notion of them being equal would mean that we basically once again remove one of the validation cases we have:
```js
case !adapter.utils.isValid(value):
  return 'invalidDate';
```

Reply:
`null` is not an invalid value, so for me yes we should fire when changing from `null` to `Invalid Date`

Reply:
Behavior updated :+1:

Reply:
I do really like the current behavior! ❤️ 
Will do some more in-depth testing, but IMHO it's a huge improvement. 👍

Reply:
It would be nice to have an "incompleteDate" error instead of "invalidDate".
But that will require deeper changes to the validation because the validation itself can't deduce this piece of information on its own...

Reply:
Sure, but I think that we firstly need to validate this behaviour and see if we'd get requests to not have this intermediate (`invalidDate`) step.
Like some users suggested, we might need to add `updateStrategy` or something along those lines to cater to users wanting more simple behaviour like **Spectrum** and maybe leave this one as default. 🤔

Reply:
A good point with the `incompleteDate` error is that you can let people ignore those specific errors.
Which would keep the firing behavior simple: we have an `onChange` that fires often but people can easily ignore some calls.
The only non-covered use-case I see is if some people want `onChange` to be called for every section change (what I initially did in this PR).

But I don't think it's worth adding a notion of "update strategy" yet.